강원대 FE 유다연 - 2단계 선물하기 메인 페이지 UI 구현하기

daaoooy opened the pull request:
# 2단계 구현 사항

- [x] 본인만의 기준으로 일관된 코드를 작성하기
- [x] 기능 단위로 나누어 커밋 하기
- [x] emotion에 Color, Typography 토큰을 설정하기 (토큰 가이드)
- [x] Mobile First Design 기반의 Layout을 구현하기 (강의 사이트 설명 참조)
- [x] 상단 네비게이션 바 영역을 구현하기
- [x] 카테고리 섹션을 구현하기
- [x] 기타 섹션 (선물한 친구 선택, 카테캠 화이팅) UI를 구현하기
- [x] 실시간 급상승 선물랭킹을 구현하기

daaoooy committed:
docs(readme): 1단계 설명 및 구현할 기능 목록 정리

미션1 1단계에서 요구하는 사항과 기능을 README에 작성함

daaoooy committed:
chore(deps): Typescript, prettier, emotion 패키지 설치

Typescript, prettier, emotion 패키지 설치 후 의존성 파일 업데이트됨.

daaoooy committed:
style(css): App.css와 index.css에서 불필요한 내용 제거

필요 없는 스타일 코드 삭제하고 빈 파일 유지함.

daaoooy committed:
chore(prettier): prettier 설정 및 ignore 파일 추가

prettier 설정 파일과 ignore 파일 추가함.

daaoooy committed:
refactor(app): 불필요한 코드 삭제 및 테스트 문구 작성

App.tsx와 main.tsx 의 불필요한 코드를 제거하고 테스트 문구를 작성함

daaoooy committed:
chore(config): 절대 경로 alias 설정 추가

tsconfig.app.json과 vite.config.ts에 절대 경로 import를 위한 alias를 설정함.

daaoooy committed:
style(index): pretendard 웹폰트 link 태그 추가

global.ts에 emotion을 사용해 전역 스타일링을 정의하고 pretendard 폰트를 적용함. \nindex.html에 pretendard 폰트 적용을 위한 link 태그를 추가함.

daaoooy committed:
chore(assets): 불필요한 SVG 파일 제거

vite.svg와 react.svg 파일은 필요 없어서 삭제함.

daaoooy committed:
docs: 2단계 요구사항 작성

daaoooy committed:
chore: 패키지 및 빌드 업데이트

daaoooy committed:
feat: Mobile Fisrt Design layout 구현

daaoooy committed:
feat: 디자인 토큰 생성

daaoooy committed:
feat: svg 파일을 불러오기 위한 기능 추가

daaoooy committed:
feat: 추가

daaoooy committed:
feat: 카테고리 섹션 기능

daaoooy committed:
feat: 기타 섹션- 코멘트 섹션

daaoooy committed:
feat: 선물할 친구 선택 섹션

daaoooy committed:
feat: 상단 네비게이션 바

daaoooy committed:
feat: 실시간 급상승 랭킹 섹션

daaoooy committed:
feat: svg 파일을 불러오기 위한 기능 추가

daaoooy committed:
refactor: 디자인 토큰 적용

daaoooy committed:
refactor: 디자인 토큰 적용 및 컴포넌트 적용

daaoooy committed:
chore: 불필요한 파일 삭제

daaoooy committed:
rename: 파일 이름 변경

daaoooy committed:
fix: resolve merge conflict with upstream daaoooy branch

Leejin-Yang commented on package.json:
```diff
@@ -27,6 +27,7 @@
     "typescript-eslint": "^8.30.1",
-    "vite": "^6.3.5"
+    "vite": "^6.3.5",
+    "vite-plugin-svgr": "^4.3.0"
```
이 플러그인은 어떤 용도일까요?

daaoooy replied:
> 이 플러그인은 어떤 용도일까요?

svg 파일을 img 태그로 불러올까 하다가 방법을 찾아보던 중, svg 파일을 컴포넌트로 불러오면 svg가 제공하는 css 속성을 사용할 수 있다는 것을 보았습니다. 현재 비록 뒤로가기와 프로필 아이콘이긴 하지만, 한 번 사용해보는 것이 좋을 것 같아 진행해보는 과정에서 설치한 플러그인입니다! 혹시 그냥 img 태그를 사용하는 것이 좋을까요?

Leejin-Yang replied:
아이콘은 주로 svg를 사용하고 있어요.
svg를 사용했을 때 이미지가 깨지지 않고, 말씀 해주신 것처럼 css 속성을 사용할 수 있어서 큰 이점이 있습니다 😊

Leejin-Yang commented on tsconfig.app.json:
```diff
@@ -23,11 +23,18 @@
     "noFallthroughCasesInSwitch": true,
     "noUncheckedSideEffectImports": true,
 
+    "baseUrl": ".",
+    "paths": {
+      "@components/*": ["src/components/*"],
+      "@/*": ["src/*"]
+    }
+    "noUncheckedSideEffectImports": true,
```
중복 설정이 있네요. 확인해주세요!

daaoooy replied:
> 중복 설정이 있네요. 확인해주세요!

충돌 해결하다 문제가 생겼나봅니다.. > 수정 완료

daaoooy replied:
> 중복 설정이 있네요. 확인해주세요!

충돌을 해결하는 과정에서 오류가 난 것 같습니다. 수정 완료했습니다!

Leejin-Yang commented on src/component/CategorySection.tsx:
```diff
@@ -0,0 +1,41 @@
+/** @jsxImportSource @emotion/react */
```
모든 컴포넌트 파일에서 해당 주석을 작성해야하는 번거로움이 있는데요, 이러한 작업을 줄이기 위해서 어떻게 할 수 있을까요?

[참고] [Emotion jsx-pragma](https://emotion.sh/docs/css-prop#jsx-pragma)

daaoooy replied:
> 모든 컴포넌트 파일에서 해당 주석을 작성해야하는 번거로움이 있는데요, 이러한 작업을 줄이기 위해서 어떻게 할 수 있을까요?
> 
> [참고] [Emotion jsx-pragma](https://emotion.sh/docs/css-prop#jsx-pragma)

-> 수정 완료

Leejin-Yang commented on src/styles/emotion.d.ts:
```diff
@@ -0,0 +1,6 @@
+import '@emotion/react'
+import { ThemeType } from '@/styles/theme'
+
+declare module '@emotion/react' {
+  export interface Theme extends ThemeType {}
+}
```
타입 정의 잘 해주셨네요 👍

Leejin-Yang commented on src/component/TrendingGiftRanking.tsx:
```diff
@@ -0,0 +1,130 @@
+            <img src={item.imageURL} alt={item.name} css={imgStyle} />
+            <p>{item.brandInfo.name}</p>
+            <p>{item.name}</p>
+            <strong>{item.price.sellingPrice.toLocaleString()} 원</strong>
```
LocaleString도 적용해주셨네요 👍

Leejin-Yang commented on src/component/CategorySection.tsx:
저는 컴포넌트 가독성을 위해 스타일 관련 코드들은 컴포넌트 아래에 작성하곤 해요. 또는 스타일 관련 코드는 따로 분리하곤 합니다. 참고해서 고민해보면 좋을 것 같아요 😊

daaoooy replied:
> 저는 컴포넌트 가독성을 위해 스타일 관련 코드들은 컴포넌트 아래에 작성하곤 해요. 또는 스타일 관련 코드는 따로 분리하곤 합니다. 참고해서 고민해보면 좋을 것 같아요 😊

-> 수정 완료 (스타일 코드 분리)

Leejin-Yang commented on src/component/CategorySection.tsx:
```diff
@@ -0,0 +1,41 @@
+import type { ThemeType } from '@/styles/theme'
+
+const containerStyle = (theme: ThemeType) => css`
+  margin: 16px 0px;
```
작성한 spacing 토큰을 활용해 여백 값들을 작성하는건 어떨까요?

daaoooy replied:
> 작성한 spacing 토큰을 활용해 여백 값들을 작성하는건 어떨까요?

-> 수정 완료

Leejin-Yang commented on src/component/CategorySection.tsx:
```diff
@@ -0,0 +1,41 @@
+
+  return (
+    <section css={containerStyle(theme)}>
+      <h2 css={{ margin: 0, fontSize: 16, fontWeight: 600 }}>선물 테마</h2>
```
몇몇 태그들은 인라인으로 작성했는데 다연님만의 기준이 있을까요?
스타일은 따로 정의해서 관리하는게 좋아보이는데 어떻게 생각하시나요?

daaoooy replied:
> 몇몇 태그들은 인라인으로 작성했는데 다연님만의 기준이 있을까요? 스타일은 따로 정의해서 관리하는게 좋아보이는데 어떻게 생각하시나요?

-> 맞습니다. 여러가지 측면에서 스타일을 따로 정의해서 관리하는 것이 좋다고 생각합니다. 몇몇 태그들에 인라인으로 적혀있는 부분은 저만의 기준이라기 보다는, 테스트하면서 빠르게 확인해보려고 썼던 부분입니다만..ㅜㅜ 마감 기한까지 코드 정리가 안 돼서 몇몇 태그들이 저런 상태로 제출해버렸습니다.. 이제부턴 바로 바로 스타일을 분리하면서 적용해야 할 것 같습니다. 인라인으로 표현한 부분 스타일로 따로 정의해주었습니다.

Leejin-Yang commented on src/component/TopNavigationBar.tsx:
```diff
@@ -0,0 +1,36 @@
+  ${theme.typography.title1Bold};
+`
+
+const TopNavigationBar: React.FC = () => {
```
FC 타입을 따로 작성한 이유가 있을까요?

Leejin-Yang replied:
> 이 부분은 타입스크립트를 너무 억지로 쓰려고 한 나머지 발생한(?) 코드인데.. (https://kg-dlife.tistory.com/220 참고했음 ..) 이렇게 쓸 필요는 없는 거겠죠? 하면서도 이게 맞나 싶었어요. 지우는 것이 좋을까요?

이 부분에도 정답은 없어보여요! FC 타입이 어떻게 생겨왔고 변해왔는지 한번 찾아보는 것도 좋아보여요. 그리고 커뮤니티에서도 관련 내용을 다루는 내용이 많은데요, 참고할만한 [링크](https://github.com/payloadcms/payload/discussions/7513) 남겨둘게요. 참고해보시고 다연님만의 스타일을 찾아가면 될 것으로 보입니다 👍

Leejin-Yang commented on src/component/TrendingGiftRanking.tsx:
```diff
@@ -0,0 +1,130 @@
+  border-radius: 8px;
+`
+
+const ButtonListContainer = (theme: ThemeType) => css`
+  display: flex;
+  width: 100%;
+  border-radius: 8px;
+  overflow: hidden;
+  border: 1px solid ${theme.colors.colorScale.blue[300]};
+  margin: 0 auto 20px;
+`
+
+const Button = (theme: ThemeType) => css`
+  flex: 1;
+  padding: 10px 0;
+  background-color: ${theme.colors.colorScale.blue[100]};
+  color: ${theme.colors.colorScale.blue[500]};
+  border: none;
+  cursor: pointer;
+  font-weight: 400;
+`
```
이 스타일들은 다른 스타일 네이밍과 동일한 컨벤션을 가져가면 더 좋아보여요!

daaoooy replied:
> 이 스타일들은 다른 스타일 네이밍과 동일한 컨벤션을 가져가면 더 좋아보여요!

-> 수정 완료

Leejin-Yang commented on src/data/products.ts:
```diff
@@ -0,0 +1,147 @@
+export interface Price {
+  basicPrice: number
+  discountRate: number
+  sellingPrice: number
+}
+
+export interface BrandInfo {
+  id: number
+  name: string
+  imageURL: string
+}
+
+export interface Product {
+  id: number
+  name: string
+  imageURL: string
+  price: Price
+  brandInfo: BrandInfo
+}
```
인터페이스를 잘 나누어서 작성해주셨네요 👍
현재 타입을 데이터 파일에 같이 정의하고 있는데, 구현하면서 여러 곳에서 쓰일 가능성이 있다면 타입을 따로 관리하는 것도 좋아보여요

Leejin-Yang commented on src/component/TrendingGiftRanking.tsx:
```diff
@@ -0,0 +1,130 @@
+  font-weight: 400;
+`
+
+const FilterItem: React.FC<{ icon: string; label: string }> = ({
+  icon,
+  label,
+}) => (
+  <div css={selectItemStyle}>
+    <div css={selectButtonStyle}>{icon}</div>
+    <div css={selectLabelStyle}>{label}</div>
+  </div>
+)
```
이 컴포넌트를 분리하는건 어떨까요?

daaoooy replied:
> 이 컴포넌트를 분리하는건 어떨까요?

-> 수정 완료 (FIlter 들 따로 파일 분리)

Leejin-Yang commented on src/styles/theme.ts:
```diff
@@ -0,0 +1,237 @@
+export const theme = {
```
테마를 잘 정리해주셨네요 👍
한가지 더 나아가자면, 컬러, 타이포, 스페이싱은 서로 다른 역할을 하고 있는데요, 각각을 분리해서 관리하면 가독성이나 유지보수성을 높여줄 수 있어 보여요 😊

daaoooy replied:
> 테마를 잘 정리해주셨네요 👍 한가지 더 나아가자면, 컬러, 타이포, 스페이싱은 서로 다른 역할을 하고 있는데요, 각각을 분리해서 관리하면 가독성이나 유지보수성을 높여줄 수 있어 보여요 😊

-> 토큰 역할별로 나누기 완료

Leejin-Yang commented on src/styles/theme.ts:
```diff
@@ -0,0 +1,237 @@
+export const theme = {
+  colors: {
+    /* 컬러 스케일 (Color Scale) */
+    colorScale: {
```
color scale과 semantic color를 잘 분리해주셨네요 👍

Leejin-Yang reviewed (changes requested):
다연님 안녕하세요!
프론트엔드 개발의 시작점에 오셨네요 🙌 UI 구현하느라 고생하셨어요!

몇가지 리뷰 남겨두었는데 확인 부탁드려요.
그리고 추가적으로 구현해야 할 부분 해주시고 리뷰 요청주세요 😊

---

코드를 잘 구현하는 것도 물론 중요하지만, 주어진 시간 안에 요구사항을 구현하는 것이 더 중요하다고 생각해요(저도 아직 그 딜레마에 갇혀 마감시간에 쫓기곤 하지만요 😅). 미션을 진행하면서 어려운 부분도 있을 것이지만, 요구사항을 만족하는 기능 구현을 우선 순위로 생각해보면 좋아보여요. 리뷰를 받고 피드백을 반영해보는 시간이 또 있으니까요 😊

궁금한 점은 언제든지 코멘트와 디스코드 DM으로 남겨주세요

daaoooy committed:
feat(etc) : 기타 섹션(코멘트 부분) 파일 이름 수정, 스타일 분리, UI 구현 완료

- 기타 섹션(코멘트 부분) 의 파일 이름을 수정함\n- 기타섹션(코멘트 부분)의 스타일을 따로 파일(EtceteraSection.styles.ts)에 분리함\n- UI 구현 완료

daaoooy committed:
feat(category) : 카테고리 파일 이름 수정, 스타일 분리, UI 및 상태 관리 구현 완료

- 카테고리의 파일 이름을 수정함\n- 카테고리의 스타일을 따로 파일(GiftCategory.styles.ts)에 분리함\n- UI 및 상태 관리 구현 완료

daaoooy committed:
feat(etc) : 기타 섹션(선물할 친구) 파일 이름 수정, 스타일 분리, UI 구현 완료

- 기타 섹션(선물할 친구)의 파일 이름을 수정함\n- 기타 섹션(선물할 친구)의 스타일을 따로 파일(GiftRecipient.styles.ts)에 분리함\n- UI 구현 완료

daaoooy committed:
feat(login) : 로그인 페이지 기능 파일 및 스타일 파일 생성

- 3단계 진행을 위해 로그인 페이지 파일 생성

daaoooy committed:
feat(top-nav) : 상단 네비게이션 바 스타일 분리, UI 구현 완료

- 상단 네비게이션 스타일을 따로 파일(TopNavigationBar.styles.ts)에 분리함\n- 상단 네비게이션 바의 뒤로가기 및 프로필에 대한 기능과 상태는 3단계를 진행하면서 구현할 예정

daaoooy committed:
feat(rank) : 실시간 급상승 선물 랭킹 스타일 분리, UI 및 상태 관리 구현 완료

- 실시간 급상승 선물 랭킹 스타일을 따로 파일(TrendingGiftRanking.styles.ts)에 분리함\n- UI 및 상태 관리 구현 완료\n- Filter 컴포넌트를 분리

daaoooy committed:
chore(data) : product 데이터 테스트용 개수 조정

- 테스트 및 더보기 기능 구현을 위해 상품 개수를 20개 정도 추가함

daaoooy committed:
style(global) : theme 관련 일부 수정 및 mobile first design layout 구현

daaoooy committed:
refactor(theme) : theme 토큰을 color, typography, spacing으로 각각 분리

daaoooy committed:
chore: 타입 선언 파일 일부 수정

- 큰 변동 사항 없음

daaoooy committed:
refactor: 컴포넌트 이름 변경에 따른 수정

- 큰 변동 사항 없음

daaoooy committed:
chore : dependencies 업데이트 및 수정

tsconfig.app.json 파일에 중복 설정 수정\n- @jsxImportSource @emotion/react 관련 피드백 사항 반영

daaoooy committed:
docs: 2단계 요구사항 작성 및 피드백 사항 작성 (답변까지 작성)

Leejin-Yang commented on src/component/EtceteraSection.tsx:
```diff
@@ -0,0 +1,12 @@
+import { Container, SubTitle, Title } from './EtceteraSection.styles'
```
스타일 컴포넌트를 import 하는 방식을 쭉 봤을 때, `named import`와 `namespace import`를 혼용해서 사용하고 있는데요. 다연님 기준에 맞게 통일해서 사용한다면 코드 스타일을 일관되게 유지할 수 있어보여요

daaoooy replied:
> 스타일 컴포넌트를 import 하는 방식을 쭉 봤을 때, `named import`와 `namespace import`를 혼용해서 사용하고 있는데요. 다연님 기준에 맞게 통일해서 사용한다면 코드 스타일을 일관되게 유지할 수 있어보여요

너무 많은 스타일 컴포넌트들이 있을 땐 `namespace import`를, 개수가 많지 않을 땐  `named import' 를 사용하였는데 통일하는 것이 좋을까요..!?

Leejin-Yang replied:
저는 하나로 통일하는 것을 추천드립니다. 
그리고 "많다"의 기준은 사람마다 다를 수 있어서, 컨벤션을 하나로 정해둔다면 협업과 유지보수에 더 도움이 될 것 같아요 😊

Leejin-Yang commented on src/component/GiftCategory.styles.ts:
```diff
@@ -0,0 +1,36 @@
+export const Grid = styled.div`
+  display: grid;
+  grid-template-columns: repeat(5, minmax(0, 1fr));
+  gap: ${theme.spacing[3]};
```
styled를 사용한다면 props 통해서 theme 값을 가져올 수 있어요. 아래와 같이 활용할 수 있어보이네요 😊

```suggestion
  gap: ${({ theme }) => theme.spacing[3]};
```

Leejin-Yang replied:
> 1. styled vs css + useTheme : 현재 요청한 코드는 styled 로 작성하였는데요, 이전 코드에서는 css + useTheme 으로 작성했거든요.. 혹시 어떤 것이 더 적합한 사용일지 궁금합니다.

질문 주신 것에서 정답은 없다고 생각해요! 다연님에게 편리한 방식으로 통일해서 작성해주시면 될 것으로 보여요. 한가지 차이점은 두 코드를 작성하면서도 느끼셨을 거 같은데, styled는 props에서 theme의 타입을 추론해 사용할 수 있어 css와 같이 타입을 따로 작성하지 않아도 된다는 점이 있겠네요 😊

Leejin-Yang commented on src/component/GiftCategory.tsx:
```diff
@@ -0,0 +1,34 @@
+    setSelectedId(id)
+  }
+
+  useEffect(() => {
+    console.log('선택된 ID:', selectedId)
+  }, [selectedId])
```
곳곳에 console.log가 있는데요, 이는 값이 잘 나오는지 확인하는 용도일까요?
불필요한 console.log는 제거해도 무방해보여요 😊

daaoooy replied:
> 곳곳에 console.log가 있는데요, 이는 값이 잘 나오는지 확인하는 용도일까요? 불필요한 console.log는 제거해도 무방해보여요 😊

아 이 부분은 그 아이템 요소? 들을 선택했을 때 잘 맞게 선택되는지 확인해보고싶어서 추가해두었어요!

Leejin-Yang commented on src/component/TrendingGiftRanking.tsx:
```diff
@@ -0,0 +1,109 @@
+import * as S from '@/component/TrendingGiftRanking.styles'
+import { FilterGender, FilterType } from '@/component/TrendingGiftRankingFilter'
+
+const genderList = [
+  { label: 'All', icon: 'ALL' },
+  { label: '남성이', icon: '👨‍🦰' },
+  { label: '여성이', icon: '👩‍🦰' },
+  { label: '청소년이', icon: '👦' },
+]
+
+const typeList = ['받고 싶어한', '많이 선물한', '위시로 받은']
+
+type GenderLabel = (typeof genderList)[number]['label']
+type TypeLabel = (typeof typeList)[number]
```
타입을 잘 뽑아주셨네요 👍
더 나아가자면 구체적인 타입을 위해 as const를 활용할 수 있어 보이는데요, 아래 이미지와 같이 확인해볼 수 있게 되어요 😊

<img width="462" alt="Screenshot 2025-06-28 at 20 02 24" src="https://github.com/user-attachments/assets/a8ac1efb-374b-4b4d-99a5-33a45413f137" />

Leejin-Yang commented on src/component/TrendingGiftRanking.tsx:
```diff
@@ -0,0 +1,109 @@
+  const [selectedGender, setSelectedGender] = useState<GenderLabel | ''>('')
+  const [selectedType, setSelectedType] = useState<TypeLabel | ''>('')
+  const [selectedProduct, setSelectedProduct] = useState<Product | null>(null)
+  const [visibleCount, setVisibleCount] = useState(6)
```
특별한 의미를 가진 숫자는 상수화를 해서 관리하는게 좋아보여요! 협업을 하거나 다시 코드를 볼 때 더 이해하기 쉬울거에요 😊

Leejin-Yang commented on src/component/TrendingGiftRankingFilter.tsx:
잘 분리해주셨네요 👍
앞으로 미션을 진행하면서 컴포넌트를 어떤 기준으로 분리하면 좋을지, 컴포넌트의 역할은 어디까지 하면 좋을지 고민하면서 진행해보시면 될 것 같아요 😊

Leejin-Yang commented on src/theme/spacing.ts:
```diff
@@ -0,0 +1,20 @@
+/* 간격 (Spacing) */
+export const spacing = {
```
> theme 폴더를 생성하여 각각 분리해주었습니다. 이런식으로 하면 괜찮을까요?

잘 반영해주셨네요 👍 더 나아가 토큰으로 정의한 값들도 as const를 활용하면 VSCode에서 값들을 쉽게 확인할 수 있을거에요

Leejin-Yang reviewed (approved):
다연님 안녕하세요!

리뷰 반영도 잘 해주시고, 요구사항도 잘 만족해서 구현해주셨네요 👍
실행 했을 때 특별한 이슈가 없어 merge 하도록 하겠습니다.

몇가지 코멘트 남겼는데요, 다음 단계 시작하면서 함께 고민해보시면 좋아보여요 😊
한가지 더 남기자면,

> 스타일 관련 코드를 따로 분리했습니다. 각각 스타일 코드들을 파일로 따로 만들어주었습니다.

이 부분도 잘 반영해주셨네요 👍 더 나아가서 앞으로 페이지가 많아지고 컴포넌트도 많아질 건데요, 그렇게 된다면 스타일 파일도 많아질 것으로 보여요. 컴포넌트와 그 컴포넌트에 관련된 스타일을 하나의 폴더에서 함께 관리한다면 좀 더 관리하기 편해질거에요. 이 부분도 한번 참고하시고 앞으로 구조를 어떻게 가져가면 좋을지 고민해보시면 좋아보입니다 😊

다음 단계에서 보아요 🙌 고생 많으셨습니다.

---

참고로 PR 설명에 피드백 반영한 점을 잘 작성해주셨는데, 다음에는 코멘트에 Reply로 달아주시면 서로 어떤 부분을 어떻게 반영했는지 대화의 흐름을 더 쉽게 확인할 수 있을 것 같아요 😊 궁금한 점이 있으면 언제든지 물어봐주세요!
